### ISSUE TITLE
TODOs & HOWTOs

### ISSUE THREAD

**Issue by zezhishao** [zezhishao/BasicTS#95]
## How-Tos
- [ ] How to **add new datasets**.
- [ ] How to **config the config file**.
- [ ] How to **inference and save the inference results**.
- [ ] How to **customize loss function**.

## ToDos
Add more baselines:
- [ ] Add DeepAR
- [ ] Add LightGBM
- [x] Add NHiTS
- [ ] Add GRU
- [x] Add STGODE
- [ ] Add SCINet
- [x] Add STSGCN❌, STFGCN❌ (They are built on MXNet)
- [ ] Add SCINet
- [ ] Add TimesNet
- [ ] Add Vanilla Transformer
- [ ] Add GPTs for Time Series

Add more datasets:
- [ ] Add M4, M5, and M6 datasets.

Others:
- [ ] Add AutoML.

**Comment by huangst21** [comment]
您好，感谢您提供的框架。关于交通预测问题，或许您可以考虑添加上PEMS07(M)和PEMS07(L)这两个数据集？

**Comment by huangst21** [comment]
> > 您好，感谢您提供的框架。关于交通预测问题，或许您可以考虑添加上PEMS07(M)和PEMS07(L)这两个数据集？
> 
> Of course, could you provide me the source of these two datasets? E.g., links or papers.

STGCN的库里包含了这两个速度数据集： https://github.com/VeritasYin/STGCN_IJCAI-18

**Comment by zezhishao** [comment]
@all-contributors
please add @zezhishao for maintenance,infra,code,bug.
please add @LMissher for code,bug.
please add @cnstark for infra.
please add @Azusa-Yuan for bug.
please add @ywoelker for bug.
please add @hlhang9527 for bug.

**Comment by littleheuer** [comment]
How to set AMP?

**Comment by zezhishao** [comment]
Currently, BasicTS does not support AMP because [these baselines](https://github.com/zezhishao/BasicTS/tree/master/baselines) don’t have many parameters.
However, I have also recently tested AMP in BasicTS to facilitate training and inference of large time series models. AMP will be added in the future (probably next version).

If you are want to use AMP, you need to init `torch.amp.autocast` and `torch.cuda.amp.GradScaler` in the `__init__` function of the [runner](https://github.com/zezhishao/BasicTS/blob/318f223727ec140333500e4f7cbbd35cdbe64506/basicts/runners/base_tsf_runner.py#L17), and then modify the `train_iters`, `val_iters`, `test`, and `backward` function.

**Comment by WaldenBlues** [comment]
It seems that STD-MAE [2024-LJCAI] have mentioned this library. Maybe it time to put that into this repo?

**Comment by zezhishao** [comment]
> It seems that STD-MAE [2024-LJCAI] have mentioned this library. Maybe it time to put that into this repo?

Hi, the STD-MAE has been added. Enjoy!
If BasicTS does help you, please give us a star and consider citing this [paper](https://arxiv.org/abs/2310.06119):

```LaTeX
@article{shao2023exploring,
   title={Exploring Progress in Multivariate Time Series Forecasting: Comprehensive Benchmarking and Heterogeneity Analysis},
   author={Shao, Zezhi and Wang, Fei and Xu, Yongjun and Wei, Wei and Yu, Chengqing and Zhang, Zhao and Yao, Di and Jin, Guangyin and Cao, Xin and Cong, Gao and others},
   journal={arXiv preprint arXiv:2310.06119},
   year={2023}
 }
```
Thanks a lot!

**Comment by wanggong101** [comment]
我现在训练好了一套参数 怎么用它去测试  没有测试的接口函数？大神可以提供一下吗
